### ISSUE TITLE
Is ego support golang:plan9-assembly code 

### ISSUE THREAD

**Issue by ParkerXie** [edgelesssys/ego#104]
Hi,my project imports some packages that  with part codes write by assembly such as [cpuid](https://github.com/klauspost/cpuid) and [sha256-simd](https://github.com/minio/sha256-simd)，I can complie success use  “ego-go build”  and "ego sign",but when I use "ego run" to run my complied program ,it crashed with the fllowing  error:
![image](https://user-images.githubusercontent.com/839681/147808624-4f26db6f-3ca8-4863-bd98-4f716b8954c0.png)
,is there  someing wrong I did

**Comment by thomasten** [comment]
Hi @ParkerXie,
Assembly is supported, but cpuid is only partly supported. klauspost/cpuid doesn't work with the current EGo version, but we'll fix this soon. Until then, you can use simulation mode (`OE_SIMULATION=1 ego run ...`).

**Comment by ParkerXie** [comment]
@thomasten Thank you for reply,I will retry when this be fixed

**Comment by thomasten** [comment]
@ParkerXie Extended cpuid support has been merged to master and will be included in the next release.

sha256-simd requires avx512, which hasn't been supported yet. I created a first patch that seems to work fine, but I'm not certain when this will be included in a release.

But if you like, you can already try by building EGo yourself:
```
git clone https://github.com/edgelesssys/ego
cd ego/dockerfiles
DOCKER_BUILDKIT=1 docker build --build-arg egotag=master --build-arg erttag=avx512 -o. - < Dockerfile.build
```
This creates an ego_..._amd64.deb file.

(SGX+avx512 requires an Icelake CPU)

**Comment by ParkerXie** [comment]
https://github.com/edgelesssys/ego.git

> @ParkerXie Extended cpuid support has been merged to master and will be included in the next release.
> 
> sha256-simd requires avx512, which hasn't been supported yet. I created a first patch that seems to work fine, but I'm not certain when this will be included in a release.
> 
> But if you like, you can already try by building EGo yourself:
> 
> ```
> git clone https://github.com/edgelesssys/ego
> cd ego/dockerfiles
> DOCKER_BUILDKIT=1 docker build --build-arg egotag=master --build-arg erttag=avx512 -o. - < Dockerfile.build
> ```
> 
> This creates an ego_..._amd64.deb file.
> 
> (SGX+avx512 requires an Icelake CPU)

@thomasten Hi,I can't create the  ego_..._amd64.deb file with the following error:
![image](https://user-images.githubusercontent.com/839681/149645407-52fe5696-f326-4fc6-bd7a-aa9b85ffd85a.png)

and if I use the default  libssl-dev  to rebuild ,show the following error:
![image](https://user-images.githubusercontent.com/839681/149646186-ab19e299-741e-4ac7-a54a-460ca182df2a.png)

**Comment by thomasten** [comment]
Sorry, I forgot to push the fixed Dockerfile with the incremented libssl-dev version, but you've already found that out yourself. Regarding the second error, I haven't encountered that one yet. There's probably some other error in the log before this one. Can you copy-paste the complete log?

**Comment by ParkerXie** [comment]
> Sorry, I forgot to push the fixed Dockerfile with the incremented libssl-dev version, but you've already found that out yourself. Regarding the second error, I haven't encountered that one yet. There's probably some other error in the log before this one. Can you copy-paste the complete log?

@thomasten Thanks a lot,it worked for me when I followed the steps  in  Dockerfile.build that you provided to build edgelessrt and  ego locally. But Maybe for network reason,I cannot build success with docker build

**Comment by thomasten** [comment]
@ParkerXie extended cpuid support is included in EGo v0.4.1.
avx512 didn't make it into this release. We'd like to [upstream the patch to Open Enclave](https://github.com/openenclave/openenclave/pull/4336), but there are concerns regarding compatibility that must be addressed first.
I created an `avx512-0.3.0` branch in edgelessrt in case you want to build a version of EGo that includes it.
